src/components/auth/Register.jsx

```diff
@@ -1,9 +1,332 @@
+  const [adultName, setAdultName] = useState('');
+  const [adultNameError, setAdultNameError] = useState('');
+  const [email, setEmail] = useState('');
+  const [emailError, setEmailError] = useState('');
+  const [confirmEmail, setConfirmEmail] = useState('');
+  const [confirmEmailError, setConfirmEmailError] = useState('');
+  const [password, setPassword] = useState('');
+  const [passwordError, setPasswordError] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [confirmPasswordError, setConfirmPasswordError] = useState('');
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const navigate = useNavigate();
+
+
+  const handleUserRegistration = async (event) => {
+    event.preventDefault();
+
+    let isDataValid = true;
+
+    if (email.trim() === '') {
+      setEmailError('Email is required');
+      isDataValid = false;
+    } else {
+      setEmailError('');
+    }
+
+    if (email !== confirmEmail) {
+      setConfirmEmailError('Emails do not match');
+      isDataValid = false;
+    } else {
+      setConfirmEmailError('');
+    }
+
+    if (password.trim() === '') {
+      setPasswordError('Password is required');
+      isDataValid = false;
+    } else {
+      setPasswordError('');
+    }
+
+    if (password !== confirmPassword) {
+      setConfirmPasswordError('Passwords do not match');
+      isDataValid = false;
+    } else {
+      setConfirmPasswordError('');
+    }
+
+    if (isAdultNameRequired && adultName.trim() === '') {
+      setAdultNameError('Adult name is required for students under 16');
+      isDataValid = false;
+    } else {
+      setAdultNameError('');
+    }
+
+    if (isDataValid) {
+      console.log('SUBMIT FORM');
+      console.log(`Create ${userRole} account`);
+
+      try {
+        let result;
+        if (userRole == 'student') {
+          let studentData = {
+            "firstName": firstName.trim(),
+            "lastName": lastName.trim(),
+            "email": email.trim(),
+            "password": password,
+            "dateOfBirth": dateOfBirth,
+            "adultName": adultName.trim(),
+          };
+          result = await registerStudent(studentData);
+        } else {
+          let teacherData = {
+            "firstName": firstName,
+            "lastName": lastName,
+            "email": email,
+            "password": password,
+          };
+          result = await registerTeacher(teacherData);
+        }
+        console.log(result);
+        if (result.status === 201) {
+          setIsModalOpen(true);
+        }
+      } catch (error) {
+        console.log('ERROR');
+        console.log(error);
+        if (error.message.includes("taken")) {
+          setEmailError(error.message);
+        }
+      }
+    }
+
+
+  }
+
+  const switchUserRole = () => {
+    if (userRole === 'student') {
+      setUserRole('teacher');
+    } else {
+      setUserRole('student');
+    }
+  }
+
+  const handleFirstNameChange = (event) => {
```

Comment:
**Strech goal**: The way state change for user information is being handled is ok. It does work but this pattern is bolstering your code. A more streamlined approach would be to have a user object as state and then using a technique called "computed property" to dynamically adjust which property of the user object is being changed.